From 9d60ed7a45175b1e8f8f4437c47ef30de543e329 Mon Sep 17 00:00:00 2001
From: Ryan Thompson <ryan.thompson@nuand.com>
Date: Mon, 12 May 2025 18:05:20 -0500
Subject: [PATCH] hwdb: add bladeRF SDR devices

Add hwdb entries for bladeRF 1.x, legacy 1.x, and 2.0 micro devices to
enable non-root access through preexisting ID_SOFTWARE_RADIO tag.

References #20561

(cherry picked from commit 951e8cb06037d5b67c0ac86f56bcbd73f4fb2d7b)

Resolves: RHEL-72701
---
 hwdb.d/70-software-radio.hwdb | 24 ++++++++++++++++++++++++
 hwdb.d/meson.build            |  1 +
 2 files changed, 25 insertions(+)
 create mode 100644 hwdb.d/70-software-radio.hwdb

diff --git a/hwdb.d/70-software-radio.hwdb b/hwdb.d/70-software-radio.hwdb
new file mode 100644
index 0000000000..c6afcffdc3
--- /dev/null
+++ b/hwdb.d/70-software-radio.hwdb
@@ -0,0 +1,24 @@
+# This file is part of systemd.
+#
+# Database for software-defined radio devices that should be accessible to the seat owner.
+#
+# To add local entries, copy this file to
+#   /etc/udev/hwdb.d/
+# and add your rules there. To load the new rules execute (as root):
+#   systemd-hwdb update
+#   udevadm trigger
+
+################
+# Nuand
+################
+# bladeRF 1.x
+usb:v2CF0p5246*
+ ID_SOFTWARE_RADIO=1
+
+# bladeRF 1.x (legacy)
+usb:v1D50p6066*
+ ID_SOFTWARE_RADIO=1
+
+# bladeRF 2.0 micro
+usb:v2CF0p5250*
+ ID_SOFTWARE_RADIO=1
diff --git a/hwdb.d/meson.build b/hwdb.d/meson.build
index b69b6d8f25..5498d486dd 100644
--- a/hwdb.d/meson.build
+++ b/hwdb.d/meson.build
@@ -34,6 +34,7 @@ hwdb_files_test = files(
         '70-mouse.hwdb',
         '70-pda.hwdb',
         '70-pointingstick.hwdb',
+        '70-software-radio.hwdb',
         '70-sound-card.hwdb',
         '70-touchpad.hwdb',
         '80-ieee1394-unit-function.hwdb')
